src/zimcheck/zimcheck.cpp

```diff
@@ -43,36 +44,37 @@
 #include "../tools.h"
 #include "checks.h"
 
-void displayHelp()
-{
-    std::cout<<"\n"
-             "zimcheck checks the quality of a ZIM file.\n\n"
-             "Usage: zimcheck [options] zimfile\n"
-             "options:\n"
-             "-A , --all             run all tests. Default if no flags are given.\n"
-             "-0 , --empty           Empty content\n"
-             "-C , --checksum        Internal CheckSum Test\n"
-             "-I , --integrity       Low-level correctness/integrity checks\n"
-             "-M , --metadata        MetaData Entries\n"
-             "-F , --favicon         Favicon\n"
-             "-P , --main            Main page\n"
-             "-R , --redundant       Redundant data check\n"
-             "-U , --url_internal    URL check - Internal URLs\n"
-             "-X , --url_external    URL check - External URLs\n"
-             "-D , --details         Details of error\n"
-             "-B , --progress        Print progress report\n"
-             "-J , --json            Output in JSON format\n"
-             "-H , --help            Displays Help\n"
-             "-V , --version         Displays software version\n"
-             "-L , --redirect_loop   Checks for the existence of redirect loops\n"
-             "-W , --threads         count of threads to utilize (default: 1)\n"
-             "examples:\n"
-             "zimcheck -A wikipedia.zim\n"
-             "zimcheck --checksum --redundant wikipedia.zim\n"
-             "zimcheck -F -R wikipedia.zim\n"
-             "zimcheck -M --favicon wikipedia.zim\n";
-    return;
-}
+static const char USAGE[] =
+R"(Zimcheck checks the quality of a ZIM file.
+
+Usage:
+  zimcheck [options] [ZIMFILE]
+
+Options:
+ -a --all             run all tests. Default if no flags are given.
+ -0 --empty           Empty content
+ -c --checksum        Internal CheckSum Test
+ -i --integrity       Low-level correctness/integrity checks
+ -m --metadata        MetaData Entries
+ -f --favicon         Favicon
+ -p --main            Main page
+ -r --redundant       Redundant data check
+ -u --url_internal    URL check - Internal URLs
+ -x --url_external    URL check - External URLs
+ -d --details         Details of error
+ -b --progress        Print progress report
+ -j --json            Output in JSON format
+ -h --help            Displays Help
+ -v --version         Displays software version
+ -l --redirect_loop   Checks for the existence of redirect loops
+ -w=<nb_thread> --threads=<nb_thread>  count of threads to utilize [default: 1]
```

Comment:
Maybe it makes sense to mention in the commit message that short options are now accepted only in lower case (or make the short options UPPER CASE in the usage string, like before, and emphasize that they are accepted only as UPPER CASE).

Reply:
There is `We lost parsing case insensitive options on the way.`.
But I can be more explicit.

I wonder why we need upper case option ?
Almost all tools I know use lower case option and never uppercase.

Reply:
> There is `We lost parsing case insensitive options on the way.`.
Oops, didn't notice it.

> 
> I wonder why we need upper case option ? Almost all tools I know use lower case option and never uppercase.

I agree. But the old usage string documented upper case short options. If zimcheck were a popular tool with usage in a lot of scripts this change won't be welcome.

Reply:
>  If zimcheck were a popular tool with usage in a lot of scripts this change won't be welcome.

I agree. @kelson42 @rgaudin is it ok for you ?

Reply:
May I ask the technical reason why we have such a move in this PR?

Reply:
Because docopt don't do case insensitive parsing.

If we want to do it, we would have to rewrite usage string as:
```
-c -C --checksum        Internal CheckSum Test
```
instead of 
```
-c --checksum        Internal CheckSum Test
```

(I haven't check that multiple shorts options for only one long option is working)

Reply:
So, if this is confirmed that lowercase args would be better from the user perspective and habits AND we have no reasonable technical alternative, then I guess we can move forward like this but we need to make a major release!

Reply:
And I have just check and it seens to NOT work.

Reply:
At least we can preserve support for upper case short options (because that's how they were documented) instead of switching to lower case options.

Reply:
Lets keep uppercase short option only then

Reply:
Done